pkg/controllers/clusteroperator_controller.go

```diff
@@ -251,7 +256,48 @@ func (r *CloudOperatorReconciler) provisioningAllowed(ctx context.Context, infra
 		return false, nil
 	}
 
-	// TODO: Add part confirming migration is finished for KCM before allowing provisioning
+	ownedByKCM, err := r.isCloudControllerOwnedByKCM(ctx)
+	if err != nil {
+		return false, err
+	}
+	if ownedByKCM {
+		// KCM resource found and it owns Cloud provider
+		klog.Infof("Kube Controller Manager still owns Cloud Controller. Skipping...")
+
+		if err := r.setStatusAvailable(ctx); err != nil {
+			klog.Errorf("Unable to sync cluster operator status: %s", err)
+			return false, err
+		}
+
+		return false, nil
+	}
 
 	return true, nil
 }
+
+func (r *CloudOperatorReconciler) isCloudControllerOwnedByKCM(ctx context.Context) (bool, error) {
+	kcm := &operatorv1.KubeControllerManager{}
+	err := r.Get(ctx, client.ObjectKey{Name: kcmResourceName}, kcm)
+	if err != nil && !errors.IsNotFound(err) {
+		klog.Errorf("Unable to retrive Kube Config Manager object: %v", err)
+
+		if err := r.setStatusDegraded(ctx, err); err != nil {
+			klog.Errorf("Error syncing ClusterOperatorStatus: %v", err)
+			return false, fmt.Errorf("error syncing ClusterOperatorStatus: %v", err)
+		}
+		return false, err
+	}
+
+	// If there is no condition, we assume that KCM owns the Cloud Controller
+	ownedByKCM := true
+
+	if kcm.Status.Conditions != nil {
+		for _, cond := range kcm.Status.Conditions {
+			if cond.Type == KCMCloudControllerOwnershipCondition {
+				ownedByKCM = cond.Status == operatorv1.ConditionTrue
```

Comment:
i couldn't quite tell from the unit tests, but is there a test case that exercises this condition?